File: bindings/java/src/main/java/systems/zlink/contracts/core/ZlinkStopwatch.java
/* SPDX-License-Identifier: MPL-2.0 */

package systems.zlink.contracts.core;

import java.time.Duration;

/** A high-resolution elapsed-time stopwatch. The caller owns this resource and must close it. */
public interface ZlinkStopwatch extends AutoCloseable {
    Duration intermediate();

    Duration stop();

    @Override
    void close();
}

File: bindings/dotnet/src/Zlink/Contracts/Core/ZlinkStopwatch.cs
// SPDX-License-Identifier: MPL-2.0

namespace Systems.Zlink;

/// <summary>
///     A high-resolution elapsed-time stopwatch.
/// </summary>
public interface IZlinkStopwatch : IDisposable, IAsyncDisposable
{
    /// <summary>
    ///     Returns the elapsed time so far, in microseconds, without stopping.
    /// </summary>
    ulong Intermediate();

    /// <summary>
    ///     Stops the stopwatch and returns the total elapsed time in microseconds.
    /// </summary>
    ulong Stop();
}